clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -27,6 +27,180 @@ using namespace clang::driver::tools;
 using namespace clang;
 using namespace llvm::opt;
 
+// Struct that relates an AOT target value with
+// Intel CPUs, Intel GPUs, AMD and NVidia GPUs.
+struct StringToOffloadArchSYCLMap {
+  const char *ArchName;
+  SYCLSupportedOffloadArchs IntelArch;
```

Comment:
Nit: Should probably be named `SYCLArch` instead of `IntelArch` since it is used for the Nvidia and AMD ones as well.